app/controllers/concerns/listings_toolkit.rb

```diff
@@ -120,11 +98,28 @@ def update
     process_after_update
   end
 
+  private
+
+  def set_and_authorize_listing
+    @listing = Listing.find(params[:id])
+    authorize @listing
+  end
+
+  def rate_limit?
+    begin
+      rate_limit!(:listing_creation)
+    rescue ::RateLimitChecker::LimitReached => e
```

Comment:
Given the name and scope of this method catching a more specific error seemed like a good choice.